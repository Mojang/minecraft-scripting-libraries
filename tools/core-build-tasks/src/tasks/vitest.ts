import { execSync } from 'child_process';

export function vitestTask() {
    return () => {
        execSync('vitest', { stdio: 'inherit' });
    };
}
