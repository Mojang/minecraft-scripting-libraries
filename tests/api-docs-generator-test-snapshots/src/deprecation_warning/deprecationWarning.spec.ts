// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { describe, it } from 'vitest';
import { runGeneratorForTest } from '../runGeneratorForTest';

describe('Deprecation Warning', () => {
    it('Properly generates documentation with a warning for deprecated APIs', () => {
        runGeneratorForTest({
            testDir: __dirname,
            generators: ['ts', 'msdocs'],
        });
    });
});
