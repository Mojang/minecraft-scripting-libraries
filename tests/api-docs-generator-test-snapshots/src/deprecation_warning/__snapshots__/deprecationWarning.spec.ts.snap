// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > # Of Files Generated 1`] = `20`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/priorscriptapi/TOC.yml 1`] = `
"  - name: deprecation-module 1.x.x
    items:
    - name: deprecation-module 1.x.x
      href: deprecation-module-1xx/deprecation-module.md
    - name: DeprecatedEnum
      href: deprecation-module-1xx/DeprecatedEnum.md
    - name: StableEnumWithDeprecatedConstant
      href: deprecation-module-1xx/StableEnumWithDeprecatedConstant.md
    - name: DeprecatedClass
      href: deprecation-module-1xx/DeprecatedClass.md
    - name: StableClassWithDeprecatedMembers
      href: deprecation-module-1xx/StableClassWithDeprecatedMembers.md
    - name: DeprecatedInterface
      href: deprecation-module-1xx/DeprecatedInterface.md
    - name: StableInterfaceWithDeprecatedProperty
      href: deprecation-module-1xx/StableInterfaceWithDeprecatedProperty.md
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/priorscriptapi/deprecation-module-1xx/DeprecatedClass.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: deprecation-module-1xx.DeprecatedClass Class
description: Contents of the deprecation-module.DeprecatedClass class (Version 1.x.x).
---
# DeprecatedClass Class (Version 1.x.x)

> [!IMPORTANT]
> This documentation is for an older version of this module.

> [!CAUTION]
> This class is deprecated and will be removed in 2.0.0.
> 
> Custom deprecation message!

This class is deprecated.
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/priorscriptapi/deprecation-module-1xx/DeprecatedEnum.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: deprecation-module-1xx.DeprecatedEnum Enumeration
description: Contents of the deprecation-module.DeprecatedEnum enumeration (Version 1.x.x).
---
# DeprecatedEnum Enumeration (Version 1.x.x)

> [!IMPORTANT]
> This documentation is for an older version of this module.

> [!CAUTION]
> This enum is deprecated and will be removed in 2.0.0.

## Constants
### **Two**
\`Two = 0\`
### **One**
\`One = 1\`
### **Three**
\`Three = 2\`
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/priorscriptapi/deprecation-module-1xx/DeprecatedError.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: deprecation-module-1xx.DeprecatedError Class
description: Contents of the deprecation-module.DeprecatedError class (Version 1.x.x).
---
# DeprecatedError Class (Version 1.x.x)

> [!CAUTION]
> This class is deprecated and will be removed in 2.0.0.

## Extends
- *Error*

## Properties

### **exampleProperty**
\`read-only exampleProperty: string[];\`

Type: *string*[]

> [!CAUTION]
> This property is deprecated and will be removed in 2.0.0.
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/priorscriptapi/deprecation-module-1xx/DeprecatedInterface.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: deprecation-module-1xx.DeprecatedInterface Interface
description: Contents of the deprecation-module.DeprecatedInterface class (Version 1.x.x).
---
# DeprecatedInterface Interface (Version 1.x.x)

> [!IMPORTANT]
> This documentation is for an older version of this module.

> [!CAUTION]
> This interface is deprecated and will be removed in 2.0.0.
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/priorscriptapi/deprecation-module-1xx/StableClassWithDeprecatedMembers.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: deprecation-module-1xx.StableClassWithDeprecatedMembers Class
description: Contents of the deprecation-module.StableClassWithDeprecatedMembers class (Version 1.x.x).
---
# StableClassWithDeprecatedMembers Class (Version 1.x.x)

> [!IMPORTANT]
> This documentation is for an older version of this module. Go to the latest documentation [*here*](../../../scriptapi/deprecation-module/StableClassWithDeprecatedMembers.md).

## Properties

### **deprecatedProperty**
\`deprecatedProperty: boolean;\`

Type: *boolean*

> [!CAUTION]
> This property is deprecated and will be removed in 2.0.0.

Notes:
  - This property can't be used in read-only mode.

### **stableProperty**
\`stableProperty: boolean;\`

Type: *boolean*

Notes:
  - This property can't be used in read-only mode.

## Methods
- [deprecatedFunction](#deprecatedfunction)
- [stableFunction](#stablefunction)

### **deprecatedFunction**
\`
deprecatedFunction(): boolean
\`

**Returns** *boolean*

> [!CAUTION]
> This function is deprecated and will be removed in 2.0.0.
  
Notes:
- This function can't be called in read-only mode.

### **stableFunction**
\`
stableFunction(): boolean
\`

**Returns** *boolean*
  
Notes:
- This function can't be called in read-only mode.

## Constants

### **deprecatedConstant**
\`static read-only deprecatedConstant: boolean;\`

Type: *boolean*

> [!CAUTION]
> This property is deprecated and will be removed in 2.0.0.

### **stableConstant**
\`static read-only stableConstant: boolean;\`

Type: *boolean*
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/priorscriptapi/deprecation-module-1xx/StableEnumWithDeprecatedConstant.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: deprecation-module-1xx.StableEnumWithDeprecatedConstant Enumeration
description: Contents of the deprecation-module.StableEnumWithDeprecatedConstant enumeration (Version 1.x.x).
---
# StableEnumWithDeprecatedConstant Enumeration (Version 1.x.x)

> [!IMPORTANT]
> This documentation is for an older version of this module. Go to the latest documentation [*here*](../../../scriptapi/deprecation-module/StableEnumWithDeprecatedConstant.md).

## Constants
### **DeprecatedVal**
\`DeprecatedVal = "deprecatedVal"\`
### **StableVal**
\`StableVal = "stableVal"\`
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/priorscriptapi/deprecation-module-1xx/StableErrorWithDeprecatedProperty.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: deprecation-module-1xx.StableErrorWithDeprecatedProperty Class
description: Contents of the deprecation-module.StableErrorWithDeprecatedProperty class (Version 1.x.x).
---
# StableErrorWithDeprecatedProperty Class (Version 1.x.x)

## Extends
- *Error*

## Properties

### **deprecatedProperty**
\`read-only deprecatedProperty: string[];\`

Type: *string*[]

> [!CAUTION]
> This property is deprecated and will be removed in 2.0.0.

### **stableProperty**
\`read-only stableProperty?: string;\`

Type: *string*
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/priorscriptapi/deprecation-module-1xx/StableInterfaceWithDeprecatedProperty.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: deprecation-module-1xx.StableInterfaceWithDeprecatedProperty Interface
description: Contents of the deprecation-module.StableInterfaceWithDeprecatedProperty class (Version 1.x.x).
---
# StableInterfaceWithDeprecatedProperty Interface (Version 1.x.x)

> [!IMPORTANT]
> This documentation is for an older version of this module. Go to the latest documentation [*here*](../../../scriptapi/deprecation-module/StableInterfaceWithDeprecatedProperty.md).

## Properties

### **deprecatedProperty**
\`deprecatedProperty: boolean;\`

Type: *boolean*

> [!CAUTION]
> This property is deprecated and will be removed in 2.0.0.

### **stableProperty**
\`stableProperty: boolean;\`

Type: *boolean*
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/priorscriptapi/deprecation-module-1xx/deprecation-module.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: deprecation-module-1xx Module
description: Contents of the deprecation-module module (Version 1.x.x)
---
# \`deprecation-module\` Module (Version 1.x.x)

> [!IMPORTANT]
> This documentation is for an older version of this module. Go to the latest documentation [*here*](../../../scriptapi/deprecation-module/deprecation-module.md).

## Manifest Details
\`\`\`json
{
    "module_name": "deprecation-module",
    "version": "1.0.0"
}
\`\`\`

## Available Versions
- \`1.1.0-beta\`
- \`1.0.0\`

## Enumerations
- [DeprecatedEnum](DeprecatedEnum.md)
- [StableEnumWithDeprecatedConstant](StableEnumWithDeprecatedConstant.md)

## Classes
- [DeprecatedClass](DeprecatedClass.md)
- [StableClassWithDeprecatedMembers](StableClassWithDeprecatedMembers.md)

## Interfaces
- [DeprecatedInterface](DeprecatedInterface.md)
- [StableInterfaceWithDeprecatedProperty](StableInterfaceWithDeprecatedProperty.md)

## Errors
- [DeprecatedError](DeprecatedError.md)
- [StableErrorWithDeprecatedProperty](StableErrorWithDeprecatedProperty.md)

## Functions

::: moniker range="=minecraft-bedrock-experimental"
### **betaGlobalFunctionNotInAlpha**
\`
betaGlobalFunctionNotInAlpha(): boolean
\`

**Returns** *boolean*

> [!CAUTION]
> This function is still in pre-release.  Its signature may change or it may be removed in future releases.
  
Notes:
- This function can't be called in read-only mode.
::: moniker-end

### **deprecatedGlobalFunction**
\`
deprecatedGlobalFunction(): boolean
\`

**Returns** *boolean*

> [!CAUTION]
> This function is deprecated and will be removed in 2.0.0.
  
Notes:
- This function can't be called in read-only mode.

### **stableGlobalFunction**
\`
stableGlobalFunction(): boolean
\`

**Returns** *boolean*

> [!CAUTION]
> This function is deprecated and will be removed in 2.0.0.
  
Notes:
- This function can't be called in read-only mode.

## Objects
  
### **stableObject**
\`static read-only stableObject: string;\`

Type: *string*
  
### **deprecatedObject**
\`static read-only deprecatedObject: string;\`

Type: *string*

> [!CAUTION]
> This property is deprecated and will be removed in 2.0.0.
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/scriptapi/TOC.yml 1`] = `
"  - name: deprecation-module
    items:
    - name: deprecation-module
      href: deprecation-module/deprecation-module.md
    - name: deprecation-module Changelog
      href: deprecation-module/changelog.md
    - name: StableEnumWithDeprecatedConstant
      href: deprecation-module/StableEnumWithDeprecatedConstant.md
    - name: StableClassWithDeprecatedMembers
      href: deprecation-module/StableClassWithDeprecatedMembers.md
    - name: StableInterfaceWithDeprecatedProperty
      href: deprecation-module/StableInterfaceWithDeprecatedProperty.md
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/scriptapi/deprecation-module/StableClassWithDeprecatedMembers.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: deprecation-module.StableClassWithDeprecatedMembers Class
description: Contents of the deprecation-module.StableClassWithDeprecatedMembers class.
---
# StableClassWithDeprecatedMembers Class

## Properties

### **stableProperty**
\`stableProperty: boolean;\`

Type: *boolean*

Notes:
  - This property can't be used in read-only mode.

## Methods
- [stableFunction](#stablefunction)

### **stableFunction**
\`
stableFunction(): boolean
\`

**Returns** *boolean*
  
Notes:
- This function can't be called in read-only mode.

## Constants

### **stableConstant**
\`static read-only stableConstant: boolean;\`

Type: *boolean*
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/scriptapi/deprecation-module/StableEnumWithDeprecatedConstant.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: deprecation-module.StableEnumWithDeprecatedConstant Enumeration
description: Contents of the deprecation-module.StableEnumWithDeprecatedConstant enumeration.
---
# StableEnumWithDeprecatedConstant Enumeration

## Constants
### **StableVal**
\`StableVal = "stableVal"\`
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/scriptapi/deprecation-module/StableErrorWithDeprecatedProperty.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: deprecation-module.StableErrorWithDeprecatedProperty Class
description: Contents of the deprecation-module.StableErrorWithDeprecatedProperty class.
---
# StableErrorWithDeprecatedProperty Class

## Extends
- *Error*

## Properties

### **stableProperty**
\`read-only stableProperty?: string;\`

Type: *string*
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/scriptapi/deprecation-module/StableInterfaceWithDeprecatedProperty.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: deprecation-module.StableInterfaceWithDeprecatedProperty Interface
description: Contents of the deprecation-module.StableInterfaceWithDeprecatedProperty class.
---
# StableInterfaceWithDeprecatedProperty Interface

## Properties

### **stableProperty**
\`stableProperty: boolean;\`

Type: *boolean*
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/scriptapi/deprecation-module/changelog.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: deprecation-module Changelog
description: Changelog of the \`deprecation-module\` module
---
# \`deprecation-module\` Changelog

## Version Changes
- [2.0.0-alpha](#200-alpha)
- [1.1.0-beta](#110-beta)
- [1.0.0](#100)

## 2.0.0-alpha
#### Removed *DeprecatedClass*
#### Changed *[\`StableClassWithDeprecatedMembers\`](StableClassWithDeprecatedMembers.md)*
- Removed property *deprecatedProperty*
- Removed constant *deprecatedConstant*
- Removed function *deprecatedFunction*
#### Removed function \`betaGlobalFunctionNotInAlpha\`
#### Removed function \`deprecatedGlobalFunction\`
#### Removed function \`stableGlobalFunction\`
#### Added function \`v2GlobalFunction\`
#### Removed enum \`DeprecatedEnum\`
#### Changed enum [\`StableEnumWithDeprecatedConstant\`](StableEnumWithDeprecatedConstant.md)
- Removed value \`DeprecatedVal\`
#### Removed object \`deprecatedObject\`
## 1.1.0-beta
#### Added function \`betaGlobalFunctionNotInAlpha\`
## 1.0.0
#### Added \`deprecation-module\` Module
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > msdocs/scriptapi/deprecation-module/deprecation-module.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: deprecation-module Module
description: Contents of the deprecation-module module
---
# \`deprecation-module\` Module

## [Changelog](changelog.md)

## Manifest Details
\`\`\`json
{
    "module_name": "deprecation-module",
    "version": "2.0.0-alpha"
}
\`\`\`
This is version 2.x.x of this module, which is the latest as of version 1.0.0 of Minecraft.

## Available Versions
- \`2.0.0-alpha\`

### Prior Versions

The following API versions are from a major version that is documented here: [*deprecation-module*](../../../priorscriptapi/deprecation-module-1xx/deprecation-module.md):
- \`1.1.0-beta\`
- \`1.0.0\`


## Enumerations
- [StableEnumWithDeprecatedConstant](StableEnumWithDeprecatedConstant.md)

## Classes
- [StableClassWithDeprecatedMembers](StableClassWithDeprecatedMembers.md)

## Interfaces
- [StableInterfaceWithDeprecatedProperty](StableInterfaceWithDeprecatedProperty.md)

## Errors
- [StableErrorWithDeprecatedProperty](StableErrorWithDeprecatedProperty.md)

## Functions

::: moniker range="=minecraft-bedrock-experimental"
### **v2GlobalFunction**
\`
v2GlobalFunction(): boolean
\`

**Returns** *boolean*

> [!CAUTION]
> This function is still in pre-release.  Its signature may change or it may be removed in future releases.
  
Notes:
- This function can't be called in read-only mode.
::: moniker-end

## Objects
  
### **stableObject**
\`static read-only stableObject: string;\`

Type: *string*
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > typescript/deprecation-module@1.0.0.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "deprecation-module",
 *   "version": "1.0.0"
 * }
 * \`\`\`
 *
 */
/**
 * @deprecated This enum is deprecated and will be removed in 2.0.0.
 */
export enum DeprecatedEnum {
    Two = 0,
    One = 1,
    Three = 2,
}

export enum StableEnumWithDeprecatedConstant {
    /**
     * @deprecated This property is deprecated and will be removed in 2.0.0.
     */
    DeprecatedVal = 'deprecatedVal',
    StableVal = 'stableVal',
}

/**
 * @deprecated This class is deprecated and will be removed in 2.0.0.
 *
 * Custom deprecation message!
 *
 * This class is deprecated.
 */
export class DeprecatedClass {
    private constructor();
}

export class StableClassWithDeprecatedMembers {
    private constructor();
    /**
     * @deprecated This property is deprecated and will be removed in 2.0.0.
     *
     * @remarks
     * This property can't be used in read-only mode.
     *
     */
    deprecatedProperty: boolean;
    /**
     * @remarks
     * This property can't be used in read-only mode.
     *
     */
    stableProperty: boolean;
    /**
     * @deprecated This property is deprecated and will be removed in 2.0.0.
     */
    static readonly deprecatedConstant: boolean;
    static readonly stableConstant: boolean;
    /**
     * @deprecated This function is deprecated and will be removed in 2.0.0.
     *
     * @remarks
     * This function can't be called in read-only mode.
     *
     */
    deprecatedFunction(): boolean;
    /**
     * @remarks
     * This function can't be called in read-only mode.
     *
     */
    stableFunction(): boolean;
}

/**
 * @deprecated This interface is deprecated and will be removed in 2.0.0.
 */
export interface DeprecatedInterface {}

export interface StableInterfaceWithDeprecatedProperty {
    /**
     * @deprecated This property is deprecated and will be removed in 2.0.0.
     */
    deprecatedProperty: boolean;
    stableProperty: boolean;
}

/**
 * @deprecated This class is deprecated and will be removed in 2.0.0.
 */
// @ts-ignore Class inheritance allowed for native defined classes
export class DeprecatedError extends Error {
    private constructor();
    exampleProperty: string[];
}

// @ts-ignore Class inheritance allowed for native defined classes
export class StableErrorWithDeprecatedProperty extends Error {
    private constructor();
    /**
     * @deprecated This property is deprecated and will be removed in 2.0.0.
     */
    deprecatedProperty: string[];
    stableProperty?: string;
}

/**
 * @deprecated This function is deprecated and will be removed in 2.0.0.
 *
 * @remarks
 * This function can't be called in read-only mode.
 *
 */
export function deprecatedGlobalFunction(): boolean;
/**
 * @deprecated This function is deprecated and will be removed in 2.0.0.
 *
 * @remarks
 * This function can't be called in read-only mode.
 *
 */
export function stableGlobalFunction(): boolean;
export const stableObject: string;
/**
 * @deprecated This property is deprecated and will be removed in 2.0.0.
 */
export const deprecatedObject: string;
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > typescript/deprecation-module@1.1.0-beta.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @beta
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "deprecation-module",
 *   "version": "1.1.0-beta"
 * }
 * \`\`\`
 *
 */
/**
 * @deprecated This enum is deprecated and will be removed in 2.0.0.
 */
export enum DeprecatedEnum {
    Two = 0,
    One = 1,
    Three = 2,
}

export enum StableEnumWithDeprecatedConstant {
    /**
     * @deprecated This property is deprecated and will be removed in 2.0.0.
     */
    DeprecatedVal = 'deprecatedVal',
    StableVal = 'stableVal',
}

/**
 * @deprecated This class is deprecated and will be removed in 2.0.0.
 *
 * Custom deprecation message!
 *
 * This class is deprecated.
 */
export class DeprecatedClass {
    private constructor();
}

export class StableClassWithDeprecatedMembers {
    private constructor();
    /**
     * @deprecated This property is deprecated and will be removed in 2.0.0.
     *
     * @remarks
     * This property can't be used in read-only mode.
     *
     */
    deprecatedProperty: boolean;
    /**
     * @remarks
     * This property can't be used in read-only mode.
     *
     */
    stableProperty: boolean;
    /**
     * @deprecated This property is deprecated and will be removed in 2.0.0.
     */
    static readonly deprecatedConstant: boolean;
    static readonly stableConstant: boolean;
    /**
     * @deprecated This function is deprecated and will be removed in 2.0.0.
     *
     * @remarks
     * This function can't be called in read-only mode.
     *
     */
    deprecatedFunction(): boolean;
    /**
     * @remarks
     * This function can't be called in read-only mode.
     *
     */
    stableFunction(): boolean;
}

/**
 * @deprecated This interface is deprecated and will be removed in 2.0.0.
 */
export interface DeprecatedInterface {}

export interface StableInterfaceWithDeprecatedProperty {
    /**
     * @deprecated This property is deprecated and will be removed in 2.0.0.
     */
    deprecatedProperty: boolean;
    stableProperty: boolean;
}

/**
 * @deprecated This class is deprecated and will be removed in 2.0.0.
 */
// @ts-ignore Class inheritance allowed for native defined classes
export class DeprecatedError extends Error {
    private constructor();
    exampleProperty: string[];
}

// @ts-ignore Class inheritance allowed for native defined classes
export class StableErrorWithDeprecatedProperty extends Error {
    private constructor();
    /**
     * @deprecated This property is deprecated and will be removed in 2.0.0.
     */
    deprecatedProperty: string[];
    stableProperty?: string;
}

/**
 * @beta
 * @remarks
 * This function can't be called in read-only mode.
 *
 */
export function betaGlobalFunctionNotInAlpha(): boolean;
/**
 * @deprecated This function is deprecated and will be removed in 2.0.0.
 *
 * @remarks
 * This function can't be called in read-only mode.
 *
 */
export function deprecatedGlobalFunction(): boolean;
/**
 * @deprecated This function is deprecated and will be removed in 2.0.0.
 *
 * @remarks
 * This function can't be called in read-only mode.
 *
 */
export function stableGlobalFunction(): boolean;
export const stableObject: string;
/**
 * @deprecated This property is deprecated and will be removed in 2.0.0.
 */
export const deprecatedObject: string;
"
`;

exports[`Deprecation Warning > Properly generates documentation with a warning for deprecated APIs > typescript/deprecation-module@2.0.0-alpha.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @alpha
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "deprecation-module",
 *   "version": "2.0.0-alpha"
 * }
 * \`\`\`
 *
 */
export enum StableEnumWithDeprecatedConstant {
    StableVal = 'stableVal',
}

export class StableClassWithDeprecatedMembers {
    private constructor();
    /**
     * @remarks
     * This property can't be used in read-only mode.
     *
     */
    stableProperty: boolean;
    static readonly stableConstant: boolean;
    /**
     * @remarks
     * This function can't be called in read-only mode.
     *
     */
    stableFunction(): boolean;
}

export interface StableInterfaceWithDeprecatedProperty {
    stableProperty: boolean;
}

// @ts-ignore Class inheritance allowed for native defined classes
export class StableErrorWithDeprecatedProperty extends Error {
    private constructor();
    stableProperty?: string;
}

/**
 * @alpha
 * @remarks
 * This function can't be called in read-only mode.
 *
 */
export function v2GlobalFunction(): boolean;
export const stableObject: string;
"
`;
